### PR TITLE
support retry-options for retry step in external solution test

### PR DESCRIPTION
**Pull request by k15r**
**Description**

Changes proposed in this pull request:

- the retry step only used default retryoptions. now those are configurable

**Related issue(s)**

### CODE REVIEW

File: tests/end-to-end/external-solution-integration/testdefinition.ko.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+kind: TestDefinition
+metadata:
+  annotations:
+    meta.helm.sh/release-name: core
+    meta.helm.sh/release-namespace: kyma-system
+  generation: 1
+  labels:
+    app: core-test-external-solution
+    app.kubernetes.io/instance: core
+    app.kubernetes.io/managed-by: Helm
```
Comment (nachtmaar) [inline tests/end-to-end/external-solution-integration/testdefinition.ko.yaml:11]: (nit) this is not managed by helm

File: tests/end-to-end/external-solution-integration/testdefinition.ko.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+apiVersion: testing.kyma-project.io/v1alpha1
+kind: TestDefinition
+metadata:
+  annotations:
+    meta.helm.sh/release-name: core
+    meta.helm.sh/release-namespace: kyma-system
+  generation: 1
```
Comment (nachtmaar) [inline tests/end-to-end/external-solution-integration/testdefinition.ko.yaml:7]: (nit) not required

File: tests/end-to-end/external-solution-integration/testdefinition.ko.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  labels:
+    app: core-test-external-solution
+    app.kubernetes.io/instance: core
+    app.kubernetes.io/managed-by: Helm
+    app.kubernetes.io/name: core-test-external-solution
+    helm.sh/chart: core-0.0.1
+    kyma-project.io/test-kind: e2e
+  name: core-test-external-solution
+  selfLink: /apis/testing.kyma-project.io/v1alpha1/namespaces/kyma-system/testdefinitions/core-test-external-solution
```
Comment (nachtmaar) [inline tests/end-to-end/external-solution-integration/testdefinition.ko.yaml:16]: (nit) not required

File: tests/end-to-end/external-solution-integration/README.md [hunk 4]
```diff
@@ -95,18 +95,11 @@ When you run the test, these actions are performed in the order listed:
...
-    ```
-    go run ./cmd/runner {test-name}
-    ```
-
-### Run against Kyma cluster in the cloud
-Run the test using the following command:
-```
-go run ./cmd/runner e2e --domain {CLUSTER_DOMAIN}
-```
+currently the test cannot be executed locally, as it requires access to cluster internal resources. To build and run the test using octopus execute
```
Comment (alexandra-simeonova) [inline tests/end-to-end/external-solution-integration/README.md:98]:
```suggestion
Currently, the test cannot be executed locally, as it requires access to cluster internal resources. To build and run the test using Octopus, execute:
```

File: tests/end-to-end/external-solution-integration/README.md [hunk 5]
```diff
@@ -95,18 +95,11 @@ When you run the test, these actions are performed in the order listed:
...
-    ```
-    go run ./cmd/runner {test-name}
-    ```
-
-### Run against Kyma cluster in the cloud
-Run the test using the following command:
-```
-go run ./cmd/runner e2e --domain {CLUSTER_DOMAIN}
-```
+Currently, the test cannot be executed locally, as it requires access to cluster internal resources. To build and run the test using Octopus, execute:
```
Comment (bszwarc) [inline tests/end-to-end/external-solution-integration/README.md:98]:
```suggestion
Currently, the test cannot be executed locally, as it requires access to internal cluster resources. To build and run the test using Octopus, execute:
```

**Pull request merged**